src/openms/source/CHEMISTRY/SvmTheoreticalSpectrumGeneratorSet.cpp

```diff
@@ -86,54 +59,56 @@ namespace OpenMS
       filename = File::find(filename);
     }
 
+    TextFile file(filename);
+    // Boost::empty()
+    if (std::begin(file) == std::end(file)) { return; }// no data to load
+
+    // skip header line
+    const auto begin_from_skipped_header = std::next(std::begin(file));
+
     Param sim_param = SvmTheoreticalSpectrumGenerator().getDefaults();
 
-    TextFile file(filename);
-    TextFile::ConstIterator it = file.begin();
+    const auto process_content = [this, filename, &sim_param] (TextFile::Iterator::value_type i)
+    {
+        std::vector<String> spl;
+        i.split(":", spl);
+        Int precursor_charge = spl[0].toInt();
 
-    if (it == file.end()) return; // no data to load
+        if (spl.size() != 2 || precursor_charge < 1)
+        {
+          throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, i, " Invalid entry in SVM model File");
+        }
+
+        //load the model into the map
+        sim_param.setValue("model_file_name", File::path(filename) + "/" + spl[1]);
+        simulators_[precursor_charge].setParameters(sim_param);
+        simulators_[precursor_charge].load();
+    };
 
-    // skip header line
-    ++it;
     // process content
-    for (; it != file.end(); ++it)
-    {
-      std::vector<String> spl;
-      it->split(":", spl);
-      Int precursor_charge = spl[0].toInt();
-
-      if (spl.size() != 2 || precursor_charge < 1)
-      {
-        throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, *it, " Invalid entry in SVM model File");
-      }
-
-      //load the model into the map
-      sim_param.setValue("model_file_name", File::path(filename) + "/" + spl[1]);
-      simulators_[precursor_charge].setParameters(sim_param);
-      simulators_[precursor_charge].load();
-    }
+    std::for_each(begin_from_skipped_header, std::end(file), process_content);
   }
 
   //Return precursor charges for which a model is contained in the set
   void SvmTheoreticalSpectrumGeneratorSet::getSupportedCharges(std::set<Size>& charges)
   {
     charges.clear();
-    std::map<Size, SvmTheoreticalSpectrumGenerator>::const_iterator it;
-    for (it = simulators_.begin(); it != simulators_.end(); ++it)
-    {
-      charges.insert(it->first);
-    }
+
+    std::transform(std::begin(simulators_), std::end(simulators_),
+                   std::inserter(charges, std::begin(charges)),
+                   [] (std::pair<Size, SvmTheoreticalSpectrumGenerator> it) { return it.first; });
```

Comment:
same here, I feel this code would be much more readable, concise and express intent to anybody
```
for (const auto& it : simulators_) charges.insert(it.first)
```

Reply:
I agree with @hroest

Reply:
The `lambda` form encapsulates the `for-loop` form. Using `std::transform` + `std::inserter` gives certain guarantees and intent about the datatype for `simulators_` and `charges` that are not promised in the `for-loop` form.
Technically this should be a `const std::pair<X,Y>` (or `const auto` in c++14), but mainly this commit was to check if this style of PR refactoring would be accepted into the codebase.
n.b. The form prepares for later c++ standards for parallel standard library algorithms.

Reply:
An author familiar wrote about this exact topic for their book in 2001[[1]](http://www.drdobbs.com/stl-algorithms-vs-hand-written-loops/184401446).
 Instead, `lambdas` superseded `std::bind2nd` in c++11 standards and beyond.


1: http://www.drdobbs.com/stl-algorithms-vs-hand-written-loops/184401446